covquestions-js/src/questionnaireEngine.ts

```diff
@@ -143,13 +160,34 @@ export class QuestionnaireEngine {
   }
 
   public getProgress(): number {
-    return (this.getCurrentQuestionIndex() + 1) / this.questions.length;
+    const currentId = this.getCurrentQuestionId();
```

Comment:
Should we rename getCurrentQuestionId with getLastAnsweredQuestionId ?
So, in general current => lastAnswered? And we keep the naming "nextQuestion". Because current is kind of misleading, is it the one currently shown or currently answered.

Reply:
Also, if we follow my suggestion in the other comment with the progress definition, we can replace this function and remove the getNextQuestion function.

  public getProgress(): number {
    const nextQuestion = this.nextQuestion();
    return nextQuestion ? this.getQuestionIndex(nextQuestion.id) / this.questions.length : 1;
  }

Reply:
I will try it out. How about the `nextUnansweredQuestion` as variable?

Reply:
We definitely need to check for nextQuestion because we need to check if the last Question is enabled or not.